hooks/shotgun_launch_publish.py

```diff
@@ -77,37 +78,119 @@ def execute(self, path, context, associated_entity, **kwargs):
         # if we return False, the app may try other ways to launch the file.
         return status
 
+    def _do_software_launcher_launch(self, path, engine_instance_name):
+        """
+        Attempts to find a Software-entity-style launcher and uses that to
+        launch if one is found.
 
-    def _do_launch(self, launch_app_instance_name, engine_name, path, context):
+        :param str path: The path to the file to open after launch.
+        :param str engine_instance_name: The name of the engine instance to
+            bootstrap.
+
+        :raises: RuntimeError when a usable launcher isn't found.
         """
-        Tries to create folders then launch the publish.
+        engine = self.parent.engine
+        launchapp_system_name = "tk-multi-launchapp"
+
+        # The apps are keyed by app instance name. We don't actually care
+        # what the specific instance is called, as we just want some instance
+        # of tk-multi-launchapp. Because of that, we need to check more than
+        # just whether the name of the app is a key in the engine's apps
+        # property.
+        launchapp_commands = []
+
+        for command_name, command_data in engine.commands.iteritems():
+            props = command_data["properties"]
+            app = props.get("app")
+            if app is not None and app.name == launchapp_system_name:
+                if props.get("engine_name") == engine_instance_name:
+                    launchapp_commands.append(command_data)
+
+        if not launchapp_commands:
+            raise RuntimeError(
+                "Unable to find an instance of %s currently running!" % launchapp_system_name
+            )
+
+        # Check to see if there's a group default. Use that if there is, and if
+        # there isn't then we take the first one off the top.
+        launch_callback = launchapp_commands[0]["callback"]
+
+        for command_data in launchapp_commands:
+            if command_data["properties"].get("group_default"):
+                launch_callback = command_data["callback"]
+                break
+
+        launch_callback(file_to_open=path)
```

Comment:
is there concern about older versions of the launch app that won't accept `file_to_open`? i.e. should this be wrapped in a `try/except`?